program/src/error.rs

```diff
@@ -0,0 +1,42 @@
+use num_derive::FromPrimitive;
+use solana_program::{
+    decode_error::DecodeError,
+    msg,
+    program_error::{PrintProgramError, ProgramError},
+};
+use thiserror::Error;
+
+#[derive(Error, Clone, Debug, Eq, PartialEq, FromPrimitive)]
+pub enum StakeError {}
+
+impl PrintProgramError for StakeError {
+    fn print<E>(&self) {
+        msg!(&self.to_string());
+    }
+}
+
+impl From<StakeError> for ProgramError {
+    fn from(e: StakeError) -> Self {
+        ProgramError::Custom(e as u32)
+    }
+}
+
+impl<T> DecodeError<T> for StakeError {
+    fn type_of() -> &'static str {
+        "Stake Error"
+    }
+}
```

Comment:
If you're interested, you can derive all of this automatically with `#[spl_program_error]`.

https://github.com/paladin-bladesmith/rewards-program/blob/7095a7f16367aa285b4b91ea686d6a1e0f027e85/program/src/error.rs#L6

Reply:
Nice!